src/engine/strat_engine/backstore/metadata.rs

```diff
@@ -36,7 +36,54 @@ pub struct BDA {
     regions: mda::MDARegions,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq)]
+enum Copy {
+    Both,
+    First,
+    Second,
+}
+
 impl BDA {
+    fn read<F>(f: &mut F) -> Result<[u8; _BDA_STATIC_HDR_SIZE]>
+    where
+        F: Read + Seek,
+    {
+        #![allow(unused_io_amount)]
+        f.seek(SeekFrom::Start(0))?;
+        let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
+
+        // FIXME: See https://github.com/stratis-storage/stratisd/pull/476
```

Comment:
I know this PR just copies this FIXME from another spot, but I went and looked at that PR and wasn't clear what should be fixed here. Should this just be doing a `read_all` to avoid short reads, or what?

Reply:
I'm not exactly sure.  I do know that we will fail the unit tests if we do a `read_exactly` as I tried that as a potential solution.

Reply:
The reason: https://github.com/stratis-storage/stratisd/pull/860#issuecomment-376331069.